general/utilities/restart_ioc_when_pv_in_alarm.py

```diff
@@ -0,0 +1,31 @@
+import sys
+import os
+sys.path.insert(0, os.path.abspath(os.path.join(os.environ["KIT_ROOT"], "ISIS", "inst_servers", "master")))
+from genie_python import genie as g
+from time import sleep
+from server_common.helpers import register_ioc_start
+from typing import List
+
+IOC_TO_RUN_IN = "BGRSCRPT_02"
+
+
+def restart_ioc_when_pv_in_alarm(block_to_monitor: str, iocs_to_restart: List[str], error_states: List[str],
+                                 wait_between_restarts: int = 120, wait_for_polling: int = 10):
+    """
+    Monitor and restart an IOC when a pv goes into a specified alarm state.
+    :param block_to_monitor: Block to monitor PV status.
+    :param iocs_to_restart: List of IOCs to restart if the block goes into an alarm.
+    :param error_states: List of error states that should cause the IOC to restart
+    :param wait_between_restarts: The time (seconds) to wait between restarts
+    :param wait_for_polling: The time (seconds) to wait between polling for the block's value
+    """
+    register_ioc_start(IOC_TO_RUN_IN)
+    g.set_instrument(None)
```

Comment:
I think these can be pulled into the background script part. E.g. see how the plotting does it https://github.com/ISISComputingGroup/ibex_developers_manual/wiki/Background-Script-IOC

Reply:
I would also maybe pull the while loop out or spawn it on a thread. As it stands this is a blocking function, which seems a bit meh